src/EnergyPlus/WeatherManager.cc

```diff
@@ -2309,33 +2309,35 @@ namespace WeatherManager {
         if (state.dataEnvrn->EMSBeamSolarRadOverrideOn) state.dataEnvrn->BeamSolarRad = state.dataEnvrn->EMSBeamSolarRadOverrideValue;
         state.dataEnvrn->LiquidPrecipitation =
             state.dataWeatherManager->TodayLiquidPrecip(state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay) / 1000.0; // convert from mm to m
-        if ((state.dataEnvrn->RunPeriodEnvironment) && (!state.dataGlobal->WarmupFlag)) {
-            int month = state.dataEnvrn->Month;
-            state.dataWaterData->RainFall.MonthlyTotalPrecInWeather.at(month - 1) += state.dataEnvrn->LiquidPrecipitation * 1000.0;
-            if ((state.dataEnvrn->LiquidPrecipitation > 0) && (state.dataGlobal->TimeStep == 1)) {
-                state.dataWaterData->RainFall.numRainyHoursInWeather.at(month - 1) += 1;
-            }
-        }
 
         // if using epw precipitation, throw warning
         if (state.dataWaterData->RainFall.ModeID == DataWater::RainfallMode::EPWPrecipitation) {
-            if ((state.dataEnvrn->LiquidPrecipitation == 0) && (state.dataEnvrn->IsRain)) {
-                state.dataEnvrn->LiquidPrecipitation = 1.5 / 1000.0;
+            if ((state.dataEnvrn->LiquidPrecipitation == 0) &&
+                // this is the Present Weather Codes field in epw. See column 1-3 in Table 2.16 Weather Codes Field Interpretation
+                (state.dataWeatherManager->TodayIsRain(state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay)) &&
+                (state.dataWeatherManager->UseRainValues)) {
+                state.dataEnvrn->LiquidPrecipitation = 1.5 / 1000.0 / state.dataGlobal->TimeStepZone;
```

Comment:
`NumOfTimeStepInHour` instead of `TimeStepZone` here. Also, the literal `1.5/1000` should be declared here as a `constexpr` variable, named something like `defaultLiquidPrecip`.

Reply:
Right. I should have it divided by NumOfTimeStepInHours. I just added the defaultLiquidPrecip to hold the constant.